engine/src/internal_development.cpp

```diff
@@ -691,31 +712,26 @@ void MCInternalObjectListenerGetListeners(MCExecContext& ctxt, MCStringRef*& r_l
 	while(t_listener != nil)
 	{
         MCStringRef t_string;
-		if (!t_listener -> object -> Exists())
-			remove_object_listener_from_list(t_listener, t_prev_listener);
-		else
-		{
-			MCObjectListenerTarget *t_target;
-			t_target = nil;
-			MCObjectListenerTarget *t_prev_target;
-			t_prev_target = nil;
-            
-            MCAutoValueRef t_long_id;
-			t_listener -> object -> Get() -> names(P_LONG_ID, &t_long_id);
-            
-			for (t_target = t_listener -> targets; t_target != nil; t_target = t_target -> next)
-			{
-				if (!t_target -> target -> Exists())
-					remove_object_listener_target_from_list(t_target, t_prev_target, t_listener, t_prev_listener);
-				else if (t_target -> target ==  t_current_object)
-                {
-                    ctxt . ConvertToString(*t_long_id, t_string);
-                    t_listeners . Push(t_string);
-                }
-			}
-			
-			t_prev_listener = t_listener;
-		}
+
+        MCObjectListenerTarget *t_target;
+        t_target = nil;
+        MCObjectListenerTarget *t_prev_target;
+        t_prev_target = nil;
+        
+        MCAutoValueRef t_long_id;
+        t_listener -> object -> Get() -> names(P_LONG_ID, &t_long_id);
```

Comment:
You cannot assume that object is non-nil, nor it exists - same for target in the sub-list. The listener structure could be locked as it is inside the dispatch process, a handler called during that process unlistens some objects, deletes some objects and then does 'get listeners'. The former two steps will cause non-existant objects and nil pointers in the structure as 'prune' cannot do anything.